service/controller/resource/statuscondition/create_test.go

```diff
@@ -0,0 +1,74 @@
+package statuscondition
+
+import (
+	"context"
+	"testing"
+
+	infrastructurev1alpha2 "github.com/giantswarm/apiextensions/v2/pkg/apis/infrastructure/v1alpha2"
+	"github.com/giantswarm/micrologger/microloggertest"
+	apiv1alpha2 "sigs.k8s.io/cluster-api/api/v1alpha2"
+
+	v1 "k8s.io/api/core/v1"
+
+	"github.com/giantswarm/cluster-operator/v3/service/internal/releaseversion"
+	tcunittest "github.com/giantswarm/cluster-operator/v3/service/internal/tenantclient/unittest"
+	"github.com/giantswarm/cluster-operator/v3/service/internal/unittest"
+)
+
+// TestComputeCreateClusterStatusConditions assumes a cluster upgrade is updating and expects updated to be set
+func TestComputeCreateClusterStatusConditions(t *testing.T) {
```

Comment:
In general it would be amazing if we could write table driven tests by default. You got a specific reason to not do that here?

Reply:
make sense, I'll change it a bit 👍